src/main/java/org/jabref/logic/importer/fileformat/BibtexParser.java

```diff
@@ -227,6 +244,8 @@ private ParserResult parseFileContent() throws IOException {
             skipWhitespace();
         }
 
+        addBibDeskGroupEntriesToJabRefGroups();
```

Comment:
Can this method be moved to a separate class?

Reason: It seems that `private Map<String, String> parsedBibdeskGroups;` is a temporary data and the data from BibDesk is moved to JabRef's internal data model.

The parsing logic is very large and maybe, we can split it up a bit.